hack/tools/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/sigstore/rekor/hack/tools
 
-go 1.23.2
+go 1.22.0
```

Comment:
why we are downgrading and not keep in go1.23?

Reply:
We're not downgrading, we are permissively declaring "this go.mod file and the module source code is API compatible with the Go standard library and toolchain from go version 1.22.0 onwards"